# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/WebCodecsUtils.h
# Commit: e903ac081227
# Full Hash: e903ac0812273c476426eba3462c4bb2977c4f2f
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 3
# Description:
#   Bug 1749047 - Move and deduplicate some utility functions to WebCodecUtils. r=chunmin
#   
#   They will be reused for the encoding side.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195699
# ==============================================================================

diff -r 029f5470314e -r e903ac081227 dom/media/webcodecs/WebCodecsUtils.h
--- a/dom/media/webcodecs/WebCodecsUtils.h	Fri Dec 22 21:39:04 2023 +0000
+++ b/dom/media/webcodecs/WebCodecsUtils.h	Fri Dec 22 21:39:05 2023 +0000
@@ -7,16 +7,17 @@
 #ifndef MOZILLA_DOM_WEBCODECS_WEBCODECSUTILS_H
 #define MOZILLA_DOM_WEBCODECS_WEBCODECSUTILS_H
 
-#include <tuple>
-
 #include "ErrorList.h"
 #include "js/TypeDecls.h"
 #include "mozilla/Maybe.h"
+#include "mozilla/MozPromise.h"
 #include "mozilla/Result.h"
-#include "mozilla/Span.h"
+#include "mozilla/TaskQueue.h"
 #include "mozilla/dom/BindingDeclarations.h"
 #include "mozilla/dom/Nullable.h"
 #include "mozilla/dom/UnionTypes.h"
+#include "mozilla/dom/VideoEncoderBinding.h"
+#include "mozilla/dom/VideoFrameBinding.h"
 
 namespace mozilla {
 
@@ -28,6 +29,12 @@
 enum class YUVColorSpace : uint8_t;
 }  // namespace gfx
 
+using WebCodecsId = size_t;
+
+extern std::atomic<WebCodecsId> sNextId;
+
+struct EncoderConfigurationChangeList;
+
 namespace dom {
 
 /*
@@ -122,7 +129,31 @@
 Maybe<VideoPixelFormat> ImageBitmapFormatToVideoPixelFormat(
     ImageBitmapFormat aFormat);
 
-}  // namespace dom
+template <typename T>
+class MessageRequestHolder {
+ public:
+  MessageRequestHolder() = default;
+  ~MessageRequestHolder() = default;
+
+  MozPromiseRequestHolder<T>& Request() { return mRequest; }
+  void Disconnect() {
+    mRequest.DisconnectIfExists();
+  }
+  void Complete() { mRequest.Complete(); }
+  bool Exists() const { return mRequest.Exists(); }
+
+ protected:
+  MozPromiseRequestHolder<T> mRequest{};
+};
+
+enum class MessageProcessedResult { NotProcessed, Processed };
+
+bool IsOnAndroid();
+bool IsOnMacOS();
+RefPtr<TaskQueue> GetWebCodecsEncoderTaskQueue();
+
+} // namespace dom
+
 }  // namespace mozilla
 
 #endif  // MOZILLA_DOM_WEBCODECS_WEBCODECSUTILS_H
